src/azure-cli/azure/cli/command_modules/acr/check_health.py

```diff
@@ -418,28 +448,157 @@ def _check_private_endpoint(cmd, registry_name, vnet_of_private_endpoint):  # py
+    if res.status_code >= 400:
+        raise CLIError(parse_error_message('Could not get the requested data.', res), res.status_code)
+    # Check if image or tag was inputted and return appropriate message
+    if res.status_code < 400 and 'sha256' not in image:
+        print_pass("Blobs referenced by tag '{}' can be pulled from registry '{}'".format(image, login_server))
+    else:
+        print_pass("Blobs in image '{}' can be pulled from registry '{}'".format(image, login_server))
 
 # General command
 def acr_check_health(cmd,  # pylint: disable useless-return
                      vnet=None,
                      ignore_errors=False,
                      yes=False,
-                     registry_name=None):
+                     registry_name=None,
+                     repository_name=None,
+                     image=None,
+                     resource_group_name=None):
     from azure.cli.core.util import in_cloud_console
+    from .repository import get_access_credentials
+    from ._utils import get_resource_group_name_by_registry_name
+
+    rg = get_resource_group_name_by_registry_name(cmd.cli_ctx, registry_name, resource_group_name)
+
+    registry = None
+
+    # Connectivity
+    try:
+        registry, _ = get_registry_by_name(cmd.cli_ctx, registry_name, rg)
+        login_server = registry.login_server.rstrip('/')
+    except CLIError:
+        from ._docker_utils import get_login_server_suffix
+        suffix = get_login_server_suffix(cmd.cli_ctx)
+
+        if not suffix:
+            from ._errors import LOGIN_SERVER_ERROR
+            _handle_error(LOGIN_SERVER_ERROR.format_error_message(registry_name), ignore_errors)
+            return
+
+    # Get the access credentials for the registry
+    login_server, username, password = get_access_credentials(
+        cmd,
+        registry_name=registry_name,
+        tenant_suffix=None,
+        username=None,
+        password=None,
+        repository=repository_name,
+        permission='pull')
+
     in_cloud_console = in_cloud_console()
     if in_cloud_console:
         logger.warning("Environment checks are not supported in Azure Cloud Shell.")
     else:
         _get_docker_status_and_version(ignore_errors, yes)
         _get_cli_version()
 
-    _check_registry_health(cmd, registry_name, ignore_errors)
+    _check_registry_health(cmd,
+                           registry_name,
+                           registry,
+                           login_server,
+                           ignore_errors)
+
+    # If repository and image are provided, validate image input
+    # Then check if blob layer can be pulled
+    if (image and repository_name):
+        validate_image(cmd,
+                       registry_name,
+                       repository_name,
+                       login_server,
+                       username,
+                       password,
+                       image)
+        _check_blob(repository_name,
```

Comment:
if both functions are only used here couldn't they be just one function that validates the image?

Reply:
@rosanch - originally had this whole thing in one huge function but was advised to break it up into methods. The validate image is just checking to make sure the image data entered is valid and then runs '_check_image_match()' to suggest image name in case there is user error/typo. The check blob is calling an API to pull a layer of the blob to see if it can be downloaded. Wanted to separate so that it's more readable and reusable.

Reply:
it's ok break it up. However, if we're always bound to both couldn't one be step inside the other instead? Because it seems every time you validate the image, you'll check the Blob as well. 
If you want to keep it as is I suggest changing the method names: _validate_image_name & _validate_image_layers